Fix/moz 370 hover states

kycapstick opened the pull request:
This PR adds hover states across the site and adjusts some of mark up so the links are padded so they take up the entire space. Broadly the rules for hover state were:

## Cards 
- Scale up by 2% and underline title (The second request in this issue: https://github.com/mozilla/community-portal/issues/496)

## Links without Underlines 
- Add underline

## Links with underline 
- Remove underline

## Inputs 
- Add blue border (As outlines in this issue: https://github.com/mozilla/community-portal/issues/347)

## Blocks 
 - Add underline to title

## Misc
- If an active state exists, mimic for hover (in places like page numbers and submenus)


Below is a detailed overview of all the places I've added hover states:

### Header

- Avatar - Underline username
- Log in & Log out - If underline exists remove it / else add it
- Search input - Blue border
- Menu Items - White after element

### Homepage

- Secondary CTA Text for groups and events - Add underline
- Card hover - Scale up by 2% and title is underlined

### Footer

- Main links - Underline
- Legal links - Remove underline
- Social links - Add underline
- Language selection - Add underline
- Languages selection - Underline language on hover

### Campaigns

- Campaign Cards - Scale up 2% and underline title

### Activities

- Related events - Add underline to title
- More events link - Add underline
- Flow titles - Add blue border
- Links in flow - Add underline
- Links in additional information sections - Add underline

### Events

- Create new link - Remove underline
- Search - Add blue border
- Filters - Add blue border
- Subnav - Blue underline

### Single Event

- Edit button - Add blue underline
- More events in this country - Add underline
- Online event only - Add underline
- Attendee card - Underline username
- External link - Add underline
- Tags - Add underline
- Activity or Campaign link - Add underline
- Group - Add underline
- Host - Add underline to username
- Language - Add underline

### Update & Create Event

- All inputs - Add blue outline

### Groups

- Verified - Remove underline
- Location - Remove underline
- Subnav links - Add underline
- Edit button - Add underline
- community links - Add underline
- Events this month - Add underline
- Members - Add underline
- Related Events - Add underline to title
- See More Events - Add underline
- Group Contact Cards -Add underline to username
- Preferred language - Add underline
- Tags - Add underline

### Group Members

- Cards - Scale up by 2% and underline title
- Search - Add blue border
- Filters - Add blue border

### People

- Edit btn - Add blue underline
- Groups - Add underline to title
- Events - Add underline to title
- Campaigns - Add underline to title
- Languages - Add underline to title
- Tags - Add underline to tag
- Social Links - Add underline

### People Edit & Create

- Inputs - Add blue outline
- Delete Button - Remove underline

### Search

- Remove underline from search result title

jveens commented:
@kycapstick Please see below!

Legend:
✅ All good!
💔 Didn't work as described
❓Not sure of result or insufficient local data to test

> ### Header
> * ✅ Avatar - Underline username
> * 💔 Log in & Log out - If underline exists remove it / else add it
> * ✅  Search input - Blue border
> * ✅ Menu Items - White after element
> 
> ### Homepage
> * ❓Secondary CTA Text for groups and events - Add underline
> * ✅ Card hover - Scale up by 2% and title is underlined
> 
> ### Footer
> * ✅ Main links - Underline
> * ✅ Legal links - Remove underline
> * ✅ Social links - Add underline
> * ❓Language selection - Add underline
> * ✅ Languages selection - Underline language on hover
> 
> ### Campaigns
> * ❓Campaign Cards - Scale up 2% and underline title
> 
> ### Activities
> * ✅ Related events - Add underline to title
> * ✅ More events link - Add underline
> * ✅ Flow titles - Add blue border
> * ✅ Links in flow - Add underline
> * ❓Links in additional information sections - Add underline
> 
> ### Events
> * ✅ Create new link - Remove underline
> * ✅ Search - Add blue border
> * ✅ Filters - Add blue border
> * ✅ Subnav - Blue underline
> 
> ### Single Event
> * ✅ Edit button - Add blue underline
> * ✅ More events in this country - Add underline
> * ❓Online event only - Add underline
> * ✅ Attendee card - Underline username
> * ✅ External link - Add underline
> * ✅ Tags - Add underline
> * ✅ Activity or Campaign link - Add underline
> * ✅ Group - Add underline
> * ✅ Host - Add underline to username
> * ✅ Language - Add underline
> 
> ### Update & Create Event
> * ✅ All inputs - Add blue outline
> 
> ### Groups
> * ❓Verified - Remove underline
> * 💔 Location - Remove underline
> * ❓Subnav links - Add underline
> * 💔 Edit button - Add underline
> * ✅ community links - Add underline
> * ✅ Events this month - Add underline
> * ❓Members - Add underline
> * ✅ Related Events - Add underline to title
> * ✅ See More Events - Add underline
> * ✅ Group Contact Cards -Add underline to username
> * ✅ Preferred language - Add underline
> * ✅ Tags - Add underline
> 
> ### Group Members
> * ✅ Cards - Scale up by 2% and underline title
> * ✅ Search - Add blue border
> * ✅ Filters - Add blue border
> 
> ### People
> * ✅ Edit btn - Add blue underline
> * ✅ Groups - Add underline to title
> * ✅ Events - Add underline to title
> * ❓Campaigns - Add underline to title
> * ✅ Languages - Add underline to title
> * ✅ Tags - Add underline to tag
> * ✅ Social Links - Add underline
> 
> ### People Edit & Create
> * ✅ Inputs - Add blue outline
> * ✅ Delete Button - Remove underline
> 
> ### Search
> * ✅ Remove underline from search result title

kycapstick commented:
Thanks for the review @jveens, I've made some changes and dropped in screenshots of everything below to help make the review process a bit clearer.

### Header

💔 Log in & Log out
Fixed. Across the site logout has an underline by default, but it was missing on the homepage due to more a more generic style.
<img width="310" alt="Screen Shot 2020-07-24 at 12 36 48 PM" src="https://user-images.githubusercontent.com/38357500/88414135-5d335100-cdaa-11ea-9afb-c63f9717bee5.png">


### Homepage

❓Secondary CTA Text for groups and events
Needs to be set in Homepage custom fields, it's fixed with hover state shown below:
<img width="255" alt="Screen Shot 2020-07-24 at 12 14 51 PM" src="https://user-images.githubusercontent.com/38357500/88412295-522af180-cda7-11ea-8846-3bb4d70a861c.png">

❓Language selection

This wording was confusing, I was referring to the language toggle in the footer, should have the hover state shown below
<img width="226" alt="Screen Shot 2020-07-24 at 12 15 34 PM" src="https://user-images.githubusercontent.com/38357500/88412359-6c64cf80-cda7-11ea-8416-1f7b5bb95511.png">


Campaigns
❓Campaign Cards 

This is the hover style for the active campaign cards, fixed hover state shown below.
<img width="690" alt="Screen Shot 2020-07-24 at 12 24 19 PM" src="https://user-images.githubusercontent.com/38357500/88413114-9ff42980-cda8-11ea-894c-0a4d342a93b2.png">


Activities

❓Links in additional information sections 

I noticed on the prod site that the Community managers are regularly adding links into the additional information text boxes, so this was a fix for those. Image below showing the hover state: 
<img width="807" alt="Screen Shot 2020-07-24 at 12 24 57 PM" src="https://user-images.githubusercontent.com/38357500/88413227-cd40d780-cda8-11ea-88d1-c1c35a9c3dd6.png">


### Events

❓Online event only 

The markup is a bit different for online events versus ones with physical locations, so needed to check links for both. Image below showing hover state:
<img width="769" alt="Screen Shot 2020-07-24 at 12 26 35 PM" src="https://user-images.githubusercontent.com/38357500/88413338-0416ed80-cda9-11ea-9f93-01d92ed09263.png">


### Groups

❓Verified 
💔 Location 

This description was not clear, it's the links just underneath the group tittle on a single group. Image below showing hover state for both (underline removed on hover):
<img width="294" alt="Screen Shot 2020-07-24 at 12 27 45 PM" src="https://user-images.githubusercontent.com/38357500/88413466-36c0e600-cda9-11ea-84c4-89bbbc716403.png">
<img width="284" alt="Screen Shot 2020-07-24 at 12 28 53 PM" src="https://user-images.githubusercontent.com/38357500/88413489-44766b80-cda9-11ea-8b0e-372913328d2b.png">

❓Subnav links 

Wasn't clear on these, didn't know what else to call them. Screen shot below (hover is on middle item): 
<img width="482" alt="Screen Shot 2020-07-24 at 12 29 32 PM" src="https://user-images.githubusercontent.com/38357500/88413560-65d75780-cda9-11ea-9de4-00083bb895c9.png">

💔 Edit button 

Pushed a fixed. Screenshot below:
<img width="220" alt="Screen Shot 2020-07-24 at 12 34 30 PM" src="https://user-images.githubusercontent.com/38357500/88413925-10e81100-cdaa-11ea-9bec-44dbfb611af9.png">

❓Members 

Wasn't clear on this one, was referring to member count in the sidebar. Screenshot below:
<img width="436" alt="Screen Shot 2020-07-24 at 12 35 12 PM" src="https://user-images.githubusercontent.com/38357500/88414023-3543ed80-cdaa-11ea-8c2c-99ab3dc96b79.png">

### Group Members

❓Campaigns 

This was a specific one, it only shows campaigns on your profile if you've used the campaign signup stuff to register for one. Screenshot of the hover state below:
<img width="852" alt="Screen Shot 2020-07-24 at 12 41 29 PM" src="https://user-images.githubusercontent.com/38357500/88414603-1db93480-cdab-11ea-98aa-fc2ac3ab2dc8.png">
